### PR TITLE
Rdm 4335 Implement and consume external POST Create Case endpoint in API v2

### PR DESCRIPTION
**Pull request by andrewfolga**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4335

### Change description ###
Implement and consume external POST Create Case endpoint in API v2

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/domain/service/createcase/ClassifiedCreateCaseOperationTest.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
     private static final String JURISDICTION_ID = "Probate";
     private static final String CASE_TYPE_ID = "Grant";
     private static final CaseDataContent EVENT_DATA = newCaseDataContent().build();
-//    private static final Map<String, JsonNode> DATA = new HashMap<>();
+    //    private static final Map<String, JsonNode> DATA = new HashMap<>();
```
Comment (andrewfolga) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/createcase/ClassifiedCreateCaseOperationTest.java]: remove

File: src/test/java/uk/gov/hmcts/ccd/domain/service/createcase/ClassifiedCreateCaseOperationTest.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
...
     private static final Boolean IGNORE = Boolean.FALSE;
 //    private static final String TOKEN = "eyvcdvsyvcdsyv";
```
Comment (andrewfolga) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/createcase/ClassifiedCreateCaseOperationTest.java]: rmeove

File: src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/CreateCaseTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    @Test
+    @DisplayName("should create case")
+    void shouldCreateCase() {
+        String eventToken = aat.getCcdHelper().generateTokenCreateCase(asAutoTestCaseworker(), JURISDICTION, CASE_TYPE, CREATE);
+
+        callCreateCase(CASE_TYPE, getBody(CREATE, eventToken, FullCase::build))
+            .when()
+            .post("/case-types/{caseTypeId}/cases")
```
Comment (rafalkalita) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/CreateCaseTest.java]: I can see plenty of API responses defined for POST /case-types/{caseTypeId}/cases, but no 201. Maybe it is worth to test other responses. Please check.

Comment (andrewfolga) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/tests/v2/external/CreateCaseTest.java]: But this tests actually tests for 201. The functional tests are here to test mainly a successful scenario and some of failure ones. All failure ones are tested in lower levels.

**Pull request merged**